hudi-utilities/src/main/java/org/apache/hudi/utilities/keygen/util/TimestampBasedKeyGeneratorHelper.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities.keygen.util;
+
+import org.apache.hudi.DataSourceUtils;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.exception.HoodieNotSupportedException;
+import org.apache.hudi.utilities.keygen.common.KeyGeneratorTimestampConfig;
+import org.apache.hudi.utilities.keygen.common.TimestampTypeEnum;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Util helps to build TimestampBasedKeyGenerator.
+ */
+public class TimestampBasedKeyGeneratorHelper {
```

Comment:
I am thinking about the naming of this class. Generally speaking, if the name of a class has a suffix like `xxxHelper` or `xxxUtils` would only contain **static** method. What about renaming to `TimestampParser` or `TimestampProcessor` or something else?

Reply:
`TimestampProcessor` seems better